src/libraries/System.Private.CoreLib/src/System/IO/StringWriter.cs

```diff
@@ -151,22 +151,133 @@ public override void Write(string? value)
 
         public override void Write(StringBuilder? value)
         {
-            if (GetType() != typeof(StringWriter))
+            // This overload was added after the Write(char[], ...) overload, and so in case
+            // a derived type may have overridden it, we need to delegate to it, which the base does.
+            if (UseBase())
             {
-                // This overload was added after the Write(char[], ...) overload, and so in case
-                // a derived type may have overridden it, we need to delegate to it, which the base does.
                 base.Write(value);
-                return;
             }
+            else
+            {
+                _sb.Append(value);
+            }
+        }
+
+        #region Write(format) + WriteLine(format)
 
+        [System.Runtime.CompilerServices.MethodImpl(Runtime.CompilerServices.MethodImplOptions.AggressiveInlining)]
```

Comment:
It would be clearer if you added a `using System.Runtime.CompilerServices;` at the beginning and remove the namespaces from the types here.